src/core/sys/posix/pthread.d

```diff
@@ -440,6 +440,144 @@ else
+    int pthread_attr_getschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_attr_init(pthread_attr_t*);
+    int pthread_attr_setdetachstate(pthread_attr_t*, int);
+    int pthread_attr_setschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_cancel(pthread_t);
+      }
+}
+else version( OpenBSD )
+{
+    // These declarations have not been checked.
+int pthread_atfork(void function(), void function(), void function());
+@nogc {
+    int pthread_atfork(void function() @nogc, void function() @nogc, void function() @nogc);
+    int pthread_attr_destroy(pthread_attr_t*);
+    int pthread_attr_getdetachstate(in pthread_attr_t*, int*);
+    int pthread_attr_getschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_attr_init(pthread_attr_t*);
+    int pthread_attr_setdetachstate(pthread_attr_t*, int);
+    int pthread_attr_setschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_cancel(pthread_t);
+      }
+}
+else version( Solaris )
+{
+    // These declarations have not been checked.
+int pthread_atfork(void function(), void function(), void function());
+@nogc {
+    int pthread_atfork(void function() @nogc, void function() @nogc, void function() @nogc);
+    int pthread_attr_destroy(pthread_attr_t*);
+    int pthread_attr_getdetachstate(in pthread_attr_t*, int*);
+    int pthread_attr_getschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_attr_init(pthread_attr_t*);
+    int pthread_attr_setdetachstate(pthread_attr_t*, int);
+    int pthread_attr_setschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_cancel(pthread_t);
+      }
+}
+else version( CRuntime_Bionic )
+{
+    // These declarations have not been checked.
+int pthread_atfork(void function(), void function(), void function());
+@nogc {
+    int pthread_atfork(void function() @nogc, void function() @nogc, void function() @nogc);
+    int pthread_attr_destroy(pthread_attr_t*);
+    int pthread_attr_getdetachstate(in pthread_attr_t*, int*);
+    int pthread_attr_getschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_attr_init(pthread_attr_t*);
+    int pthread_attr_setdetachstate(pthread_attr_t*, int);
+    int pthread_attr_setschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_cancel(pthread_t);
+      }
+}
+else version( CRuntime_Musl )
+{
+    // These declarations have not been checked.
+int pthread_atfork(void function(), void function(), void function());
+@nogc {
+    int pthread_atfork(void function() @nogc, void function() @nogc, void function() @nogc);
+    int pthread_attr_destroy(pthread_attr_t*);
+    int pthread_attr_getdetachstate(in pthread_attr_t*, int*);
+    int pthread_attr_getschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_attr_init(pthread_attr_t*);
+    int pthread_attr_setdetachstate(pthread_attr_t*, int);
+    int pthread_attr_setschedparam(in pthread_attr_t*, sched_param*);
+    int pthread_cancel(pthread_t);
+      }
+}
+else version( CRuntime_UClibc )
```

Comment:
Shouldn't C runtime versions take precedence over OS versions? Then again probably only applies realistically to glibc where there are active ports to FreeBSD and Solaris.

Reply:
Yes, you're right, that's why Glibc is always first. However, realistically we don't have to worry about that since linux is the only kernel that uses so many other libc's. If and when we support kFreeBSD or some other new libc/kernel combo, someone will have to do the work of splitting out the FreeBSD libc declarations from the kernel, which are currently bundled together, and then the order won't matter, as kFreeBSD will select for Glibc rather the separated FreeBSD libc.

Reply:
Well compilers already support kfreebsd and ksolaris _today_, so thinking about future compatibility over making it someone else's job is preferable 😉

Reply:
It's not going to work anyway, as I said, because someone will have to separate out the BSD/Solaris version blocks at that point.  No point in worrying about the order here when it's not going to ever make a difference.